### PR TITLE
FISH-8880 Upgrade Exousia to 3.0.0-M3

### PR DESCRIPTION
**Pull request by kalinchan**
## Description

Changed from the SimplePolicyProvider to the default

Uses the exousia authorization service

### CODE REVIEW

File: appserver/security/core-ee/src/main/java/com/sun/enterprise/security/web/integration/GlassFishToExousiaConverter.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import com.sun.enterprise.deployment.WebBundleDescriptor;
+import com.sun.enterprise.deployment.WebComponentDescriptor;
+import com.sun.enterprise.deployment.web.SecurityRoleReference;
+
+import jakarta.servlet.annotation.ServletSecurity.TransportGuarantee;
+
+/**
+ * This class converts from GlassFish security types to Exousia security types.
+ *
+ * @author arjan
```
Comment (breakponchito) [inline appserver/security/core-ee/src/main/java/com/sun/enterprise/security/web/integration/GlassFishToExousiaConverter.java:45]: Should we need to maintain the author name?

**Review by breakponchito — APPROVED** [review]
LGTM

File: appserver/security/core-ee/src/main/java/com/sun/enterprise/security/web/integration/GlassFishToExousiaConverter.java (hunk 1 above)
Comment (Pandrex247) [reply appserver/security/core-ee/src/main/java/com/sun/enterprise/security/web/integration/GlassFishToExousiaConverter.java:45]: If this is a port from GlassFish then yes. Would be nice to have the full name though 🤣

File: nucleus/deployment/common/src/main/java/org/glassfish/deployment/common/SecurityRoleMapper.java [hunk 2]
```diff
@@ -37,16 +37,18 @@
  * only if the new code is made subject to such option by the copyright
  * holder.
  */
+// Portions Copyright [2024] Payara Foundation and/or affiliates
```
Comment (OndroMih) [inline nucleus/deployment/common/src/main/java/org/glassfish/deployment/common/SecurityRoleMapper.java:40]: Code in this file is copied from Eclipse GlassFish. Copyright should be attributed to "Contributors to the Eclipse Foundation" or OmniFish. The copied code is licensed under EPL or GPL, it cannot be distributed under the CDDL license.

**Review by OndroMih — CHANGES_REQUESTED** [review]

**Comment by OndroMih** [comment]
Hi @kalinchan, how are you? I left a few comments where I think this PR violates license or copyright. I hope it's easy to fix.

**Comment by kalinchan** [comment]
Hi @OndroMih,

I have made the changes, let me know if you believe more are required.

**Comment by OndroMih** [comment]
Hi, @kalinchan,

Even though you added the correct copyright, both files are still distributed under both CDDL and GPL licenses. Even if you choose to use the Eclipse Foundation code under the GPL, you cannot redistribute it under a different license. So the whole file should be now only distributed under the GPL license, without CDDL.

**Comment by Pandrex247** [comment]
`// Payara Foundation and/or its affiliates elects to include this software in this distribution under the GPL Version 2 license`
With this addition it _is_ only distributed under GPL. 

```
 * If you wish your version of this file to be governed by only the CDDL or
 * only the GPL Version 2, indicate your decision by adding "[Contributor]
 * elects to include this software in this distribution under the [CDDL or GPL
 * Version 2] license."  If you don't indicate a single choice of license, a
 * recipient has the option to distribute your version of this file under
 * either the CDDL, the GPL Version 2 or to extend the choice of license to
 * its licensees as provided above.  However, if you add GPL Version 2 code
 * and therefore, elected the GPL Version 2 license, then the option applies
 * only if the new code is made subject to such option by the copyright
 * holder.
```

**Review by Pandrex247 — APPROVED** [review]
Server starts and admin console loads. I think this is enough for us to merge this initial PR, and make any additional changes incrementally.

I've created an issue, [FISH-9478](https://payara.atlassian.net/browse/FISH-9478), to review whether we should retain our `PolicyConfigurationFactoryImpl` and `PolicyProviderImpl` classes or not, as they were introduced to allow us to provide [JACC per application functionality](https://docs.payara.fish/community/docs/Technical%20Documentation/Payara%20Server%20Documentation/Security%20Guide/Administering%20System%20Security.html#how-to-install-a-custom-jacc-provider-per-application), but this seems to now be a [part of the specification itself](https://github.com/jakartaee/authorization/issues/98) and may be provided by Exousia natively.

**Pull request merged**